### PR TITLE
Add retention for prod GCS (not GCR) buckets

### PR DESCRIPTION
**Pull request by thockin**
I can't see why we would NOT want to guarantee this.  The value of 10 years was picked from the air.  It could be argued for 5y or for 100y.  5y seems so recent.

### CODE REVIEW

File: infra/gcp/ensure-prod-storage.sh [hunk 1]
```diff
@@ -138,7 +142,6 @@ empower_group_to_fake_prod \
     "k8s-infra-staging-release-test@kubernetes.io"
 
 # Special case: don't use retention on cip-test buckets
-# (the retention appears to have been locked, so we set it to 1s instead)
-gsutil retention set 1s gs://k8s-cip-test-prod
+gsutil retention clear gs://k8s-cip-test-prod
```
Comment (justinsb) [inline infra/gcp/ensure-prod-storage.sh]: OK, well I guess it wasn't locked!  Thanks for fixing :-)

**Comment by thockin** [comment]
too late :)

**Pull request merged**